apps/accounts/views.py

```diff
@@ -45,3 +45,27 @@ def get_auth_token(request):
 
     response_data = {"token": "{}".format(token)}
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication, ))
+def generate_auth_token(request):
+    try:
+        user = User.objects.get(email=request.user.email)
+    except User.DoesNotExist:
+        response_data = {"error": "This User account doesn't exist."}
+        return Response(response_data, status.HTTP_404_NOT_FOUND)
+    try:
+        old_auth_token = Token.objects.get(user_id=request.user.pk)
+    except Token.DoesNotExist:
+        response_data = {"error": "This Token account doesn't exist."}
+        return Response(response_data, status.HTTP_404_NOT_FOUND)
+    Token.delete(old_auth_token)
+    new_token = Token.objects.create(user=user)
```

Comment:
Please add a try-catch here too as it will throw an IntegrityError if the user isn't created.